test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,67 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  - interval: 1m
+    input_series:
+
+      # Pod is in the Pending state and scheduled, so it will be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Pending"}'
+        values: '1x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Unknown"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Failed"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Running"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Succeeded"}'
+        values: '0x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-1"}'
+        values: '0x15'
+
+      # Pod is in the Unknown state and unscheduled(unscheduled once in the 15 min interval), so it will not be alerted.
```

Comment:
Nice test case!

Reply:
Also add a test case in which the pod is not unschedulable but it's within the bad phase for 14 minutes rather than 15 minutes, and in a good phase for the other minute.

Something like this:
```
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Failed"}'
        values: '0 1x13'
      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Running"}'
        values: '1 0x13'
```

In this case, as far as I see it, you don't have to include all the different phases - just one good and one bad.